packages/plugin-core/src/commands/GotoNote.ts

```diff
@@ -135,27 +151,42 @@ export class GotoNoteCommand extends BasicCommand<CommandOpts, CommandOutput> {
           if (_.isUndefined(resp)) return;
           opts.vault = resp;
         } else {
-          // This is a new note. Depending on the config, we can either
-          // automatically pick the vault or we'll prompt for it.
-          const confirmVaultSetting =
-            getDWorkspace().config["lookupConfirmVaultOnCreate"];
-          const selectionMode =
-            confirmVaultSetting !== true
-              ? VaultSelectionMode.smart
-              : VaultSelectionMode.alwaysPrompt;
-
-          const currentVault = PickerUtilsV2.getVaultForOpenEditor();
-          const selectedVault = await PickerUtilsV2.getOrPromptVaultForNewNote({
-            vault: currentVault,
-            fname: opts.qs,
-            vaultSelectionMode: selectionMode,
+          // Not an existing note
+          const nonNoteFile = await findNonNoteFile({
+            fpath: opts.qs,
+            wsRoot: engine.wsRoot,
+            vaults: engine.vaults,
           });
+          if (nonNoteFile?.vault !== undefined) {
```

Comment:
can we be explicit about returning the kind of note here? so either a "note" or a "nonNote"?